src/sentry/models/event.py

```diff
@@ -320,47 +310,205 @@ def site(self):
+        # properties on self.data
+        node_id = SnubaEvent.generate_node_id(self.project_id, self.event_id)
+        self.data = NodeData(None, node_id, data=None)
+
+    # ============================================
+    # Snuba-only implementations of properties that
+    # would otherwise require nodestore data.
+    # ============================================
+    @property
+    def tags(self):
+        """
+        Override of tags property that uses tags from snuba rather than
+        the nodestore event body. This might be useful for implementing
+        tag deletions without having to rewrite nodestore blobs.
+        """
+        keys = getattr(self, 'tags.key', None)
+        values = getattr(self, 'tags.value', None)
+        if keys and values and len(keys) == len(values):
+            return sorted(zip(keys, values))
+        return []
+
+    def get_interface(self, name):
+        """
+        Override of interface getter that lets us return some interfaces
+        directly from Snuba data.
+        """
+        if name in ['user']:
+            from sentry.interfaces.user import User
+            return User.to_python({
+                'id': self.user_id,
+                'email': self.email,
+                'username': self.username,
+                'ip_address': self.ip_address,
+            })
         else:
-            template = DEFAULT_SUBJECT_TEMPLATE
-        return truncatechars(
-            template.safe_substitute(
-                EventSubjectTemplateData(self),
-            ),
-            128,
-        ).encode('utf-8')
+            return self.interfaces.get(name)
 
-    def get_environment(self):
-        from sentry.models import Environment
+    # ============================================
+    # Snuba implementations of django Fields
+    # ============================================
+    @property
+    def datetime(self):
+        """
+        Reconstruct the datetime of this event from the snuba timestamp
+        """
+        # dateutil seems to use tzlocal() instead of UTC even though the string
+        # ends with '+00:00', so just replace the TZ with UTC because we know
+        # all timestamps from snuba are UTC.
+        return parse_date(self.timestamp).replace(tzinfo=pytz.utc)
 
-        if not hasattr(self, '_environment_cache'):
-            self._environment_cache = Environment.objects.get(
-                organization_id=self.project.organization_id,
-                name=Environment.get_name_or_default(self.get_tag('environment')),
-            )
+    @property
+    def time_spent(self):
+        return None
 
-        return self._environment_cache
+    @property
+    def id(self):
+        # Because a snuba event will never have a django row id, just return
+        # the hex event_id here. We should be moving to a world where we never
+        # have to reference the row id anyway.
+        return self.event_id
```

Comment:
Is there a reason we're not returning `None` here? Do we accept this now everywhere where we used to accept the PG id?

Reply:
It should be the case that all api lookups of events can use both since https://github.com/getsentry/sentry/pull/11891 but I can return None here and see what breaks.

Reply:
I basically only track what we install on getsentry. Which sadly includes some pinned stuff that is not in getsentry